tss-esapi-sys/build.rs

```diff
@@ -1,114 +1,487 @@
+                version: version.to_string(),
+                name: lib_name.to_string(),
+            })
+        }
+
+        /// Finds the path for the lib file.
+        ///
+        /// # Arguments
+        /// `lib_name` - The name of the library to probe for.
+        /// `lib_path` - The path to probe for the library.
+        ///
+        /// # Returns
+        ///     The lib file path if it was found.
+        ///
+        /// # Panics
+        ///     - If the `lib_path` cannot be read using `read_dir`.
+        ///     - If more then one file matches the lib_name.
+        fn lib_file(lib_name: &str, lib_path: &Path) -> Option<PathBuf> {
+            let mut hit_iter = read_dir(lib_path)
+                .expect("The call to read_dir failed.")
+                .filter_map(|e| e.ok())
+                .map(|e| e.path())
+                .filter(|file| {
+                    file.extension()
+                        .and_then(|ext| ext.to_str())
+                        .map_or(false, |file_ext| ["so", "lib"].contains(&file_ext))
+                        && file
+                            .file_stem()
+                            .and_then(|stem| stem.to_str())
+                            .map_or(false, |file_name| file_name.contains(lib_name))
+                })
+                .peekable();
+
+            let result = hit_iter.next();
+            // Ensure it is a unique hit
+            if let Some(hit) = result.as_ref() {
+                if hit_iter.peek().is_some() {
+                    let mut associated_files = hit_iter.collect::<Vec<PathBuf>>();
+                    associated_files.push(hit.clone());
+                    panic!(
+                        "More then one match found for library `{}`: {:?}",
+                        lib_name, associated_files
+                    );
+                }
+            }
+            result
+        }
+
+        /// Creates a PathBuf object for the header file.
+        ///
+        /// # Args
+        /// `lib_name`          - Name of the library.
+        /// `include_path`      - The include path to the header file.
+        /// `with_header_files` - Flag indicating if header files are required.
+        ///
+        /// # Returns
+        ///     An optional PathBuf object.
+        ///
+        /// # Panics
+        ///     - If `with_header_files` but the combination of `file_name` and `include_path`
+        ///       does not point to an existing file.
+        fn header_file(
+            lib_name: &str,
+            include_path: &Path,
+            with_header_files: bool,
+        ) -> Option<PathBuf> {
+            if !with_header_files {
+                return None;
+            }
+            let file_name = PathBuf::from(lib_name.replace('-', "_"));
+            let header_file = include_path.join(file_name.with_extension("h"));
+            if with_header_files && !header_file.is_file() {
```

Comment:
Is the `with_header_files` needed here, since you already check for it at the beginning?

Reply:
No it is not necessary. I will fix that.

Reply:
Thank you!